lgc/include/lgc/state/PalMetadata.h

```diff
@@ -70,6 +72,15 @@ struct VsEntryRegInfo {
   bool wave32;                // Whether VS is wave32
 };
 
+// =====================================================================================================================
+// Struct with the information for one color export
+struct ColorExportInfo {
+  unsigned location;
+  unsigned originallocation;
+  bool isNullFs;
```

Comment:
I add this here, but it is redundant to have it for every color export.  Is there a better place to keep it?

Reply:
What is `isNullFs` needed for?

Could you document:
* What do `location` and `originalLocation` (camelCase!) mean?

Reply:
This information will be used for more than just generating the export epilog shader.  There are a couple parts of the pal metadata that need to see the pipeline state.  For example the `SPI_SHADER_COL_FORMAT` entry needs the color format in the pipeline state.  For a null fragment shader, you will need at least one color export statement for the dummy export, but the `CB_SHADER_MASK` has to be set to 0.

That is what this PR does. It generates the metadata, and uses it to set those two entries in the pal metadata.

For the comments and naming, this still needs some work. I took names from the original code.  I want to rename `location` and `originalLocation`, but I'm still not clear on a good name.  I reused the name those values were called in the original code without understand them completely.

Reply:
How about using `renderTarget` instead of `location` as in #777? Then `originalLocation` can just be `location`. Perhaps even better: `hwRenderTarget` or `hwMrt`, to clarify that it's about where things are mapped in the hardware.

On `isNullFs`: can't we just derive that when required? Basically, if the depth export is turned off and the color export array is empty, that's when we need the register settings for the special-case of the mrt0 dummy export.

Reply:
I like hwMrt.  Short and precise.

> On isNullFs: can't we just derive that when required?

I'm hope there is a way, but I cannot see how.  The pass that generates the null fragment shader generates an import and an export in the llvm-ir.  As far as I can tell, the llvm-ir could have been generated by an actual shader.  The color export array won't be empty because of the generated export.  The shader stages in the pipeline state get updated to look like the FS was there all along.  The only thing I see that sets it apart is that the `isNullFs` flag gets set in the inOutUsage.

Reply:
Looking at it closer, I am able to compute the CB_SHADER_MASK before linking.  That means I don't need to know if I have a null fs or not when linking anymore.